flume-ng-clients/flume-ng-log4jappender/pom.xml

```diff
@@ -67,6 +67,7 @@ limitations under the License.
     <dependency>
       <groupId>log4j</groupId>
       <artifactId>log4j</artifactId>
+      <version>1.2.17</version>
```

Comment:
Wouldn't be better to have this in the parent dependencyManagement as there are transitive dependencies that are still using log4j 1.x?

Reply:
No. All references to log4j 1 have been removed except for this one instance.

Reply:
All references in flume however, some dependencies of flume has it as a transitive dependency.
For example hive-shmis, hadoop-common, kafka_2.10.

We should harmonize the versions for these dependencies as well to avoid version mismatches.

Reply:
I went through and excluded all the transitive dependencies to log4j in the Flume pom.xml files that I could find. They have been replaced with references to log4j-1.2-api.

Reply:
Seeing the output of `mvn dependency:tree` there are a few places where it looks like you might have missed it. Could you please double check it to see if I am right? 

Wouldn't it be easier to use the dependencyManagement in parent pom to exclude the old version?

Reply:
It might. I will take a look again.